### PR TITLE
Implement downloader

### LINKED ISSUE #2: Create downloader package from new borges prototype
Create `downloader` package with functionality extracted from:

https://github.com/jfontan/borges/tree/new_borges

Add error control and logging. Part of the logic is in the main package.

### PR DESCRIPTION
**Pull request by mcarmonaa**
Depends on #7 
Closes #2

### CODE REVIEW

File: downloader/download.go [hunk 1]
```diff
@@ -2,13 +2,268 @@ package downloader
...
+	if len(job.Endpoints) == 0 || job.Lib == nil || job.TempFS == nil {
+		return ErrNotDownloadJob.New()
+	}
+
+	lib, ok := (job.Lib).(*siva.Library)
+	if !ok {
+		return ErrNeedSivaLibrary.New()
+	}
+
+	return downloadRepo(lib, job.TempFS, job.Endpoints[0])
```
Comment (jfontan) [inline downloader/download.go]: We should open an issue to use context for cancellation / timeout in downloads.

Comment (mcarmonaa) [reply downloader/download.go]: Issue opened at #12

File: downloader/download.go [hunk 2]
```diff
@@ -2,13 +2,268 @@ package downloader
...
+		}
+	}
+
+	if _, err := createRemote(r.R(), id, endpoint); err != nil {
+		return err
+	}
+
+	if err := r.R().Fetch(&git.FetchOptions{
+		RemoteName: id,
+		Tags:       git.NoTags,
```
Comment (jfontan) [inline downloader/download.go]: Second fetch should also download tags.

Comment (mcarmonaa) [reply downloader/download.go]: @jfontan done

**Pull request merged**